1-js/03-code-quality/04-ninja-code/article.md

```diff
@@ -1,241 +1,241 @@
-# Ninja code
+# Código ninja
 
 
-```quote author="Confucius (Analects)"
-Learning without thought is labor lost; thought without learning is perilous.
+```quote author="Confúcio"
+Estudo sem pensamento é trabalho perdido; pensamento sem estudo é perigoso.
 ```
 
-Programmer ninjas of the past used these tricks to sharpen the mind of code maintainers.
+Programadores ninjas do passado usavam truques para moldar a mente dos mantenedores de códigos.
 
-Code review gurus look for them in test tasks.
+Os gurus da revisão de código procuram por essas artimanhas em testes avaliativos.
 
-Novice developers sometimes use them even better than programmer ninjas.
+Desenvolvedores iniciantes muitas vezes usam essas habilidades até mesmo melhor que os programadores ninjas.
 
-Read them carefully and find out who you are -- a ninja, a novice, or maybe a code reviewer?
+Leia tudo com atenção e descubra quem é você -- um ninja, um iniciante ou um revisor de código?
 
 
-```warn header="Irony detected"
-Many try to follow ninja paths. Few succeed.
+```warn header="Ironia detectada"
+Muitos tentam ir pelo caminho dos ninjas. Poucos conseguem.
 ```
 
 
-## Brevity is the soul of wit
+## Brevidade é a alma da perspicácia
 
-Make the code as short as possible. Show how smart you are.
+Faça o código o mais curto possível. Mostre quão inteligente você é.
 
-Let subtle language features guide you.
+Deixe a linguagem implícita guiar você.
 
-For instance, take a look at this ternary operator `'?'`:
+Por exemplo, dê uma olhada neste operador ternário `'?'`:
 
 ```js
-// taken from a well-known javascript library
+// tirado de uma bem conhecida biblioteca javascript
 i = i ? i < 0 ? Math.max(0, len + i) : i : 0;
 ```
 
-Cool, right? If you write like that, a developer who comes across this line and tries to understand what is the value of `i` is going to have a merry time. Then come to you, seeking for an answer.
+Lega, né? Se você escrever dessa forma, um desenvolvedor que passar por essa linha e tentar entender qual o valor de `i` irá gastar um bom tempo. E então irá até você pedindo uma resposta.
 
-Tell them that shorter is always better. Initiate them into the paths of ninja.
+Diga a eles que o curto é sempre melhor. Introduza eles ao caminho do ninja.
 
-## One-letter variables
+## Variáveis de letra única
 
-```quote author="Laozi (Tao Te Ching)"
-The Dao hides in wordlessness. Only the Dao is well begun and well
-completed.
+```quote author="Lao Zi (Tao Te Ching)"
+O Caminho é invisível e não tem nome.
```

Comment:
```suggestion
O Dao se esconde na ausência de palavras. Apenas o Dao está bem iniciado e bem
concluído.
```

[Google Translate - here](https://translate.google.com/?sl=en&tl=pt&text=The%20Dao%20hides%20in%20wordlessness.%20Only%20the%20Dao%20is%20well%20begun%20and%20well%20completed.&op=translate)